src/main/java/org/opensearch/security/privileges/ActionPrivileges.java

```diff
@@ -407,6 +441,14 @@ PrivilegesEvaluatorResponse providesPrivilege(PrivilegesEvaluationContext contex
                 }
             }
 
+            // 4: If plugin is performing the action, check if plugin has permission
+            if (this.usersToActionMatcher.containsKey(context.getUser().getName())) {
```

Comment:
I didn't explicitly put the check in because `this.usersToActionMatcher` is only populated on bootstrap and we control this data structure.

While this would be limited to plugins in its current form, it could be extended to the service accounts associated with extensions. I think a check would be redundant, but we could add it in to prevent a map lookup for regular users.

Reply:
Added the check in